src/components/general-chart-components/GeneralChartComponents.js

```diff
@@ -34,23 +35,28 @@ function GeneralChartComponents({
   legendConfig = {},
   customLabelFormatter = null,
   customValueFormatter = null,
+  customNameFormatter = null,
   tooltipHandlers = {},
   isFollowUpDisabled = false,
   isSplitAxes = false,
 }) {
+  const isCartesianPlot = getIsCartesianPlot(plotConfig);
+
   return (
     <>
-      {XAxis({ ...xAxisConfig })}
-      {YAxis({ ...yAxisConfig })}
-      {YAxis({ ...secondYAxisConfig })}
-      {ZAxis({ ...zAxisConfig })}
-      {useGridLines && GridLines({})}
+      {isCartesianPlot && XAxis({ ...xAxisConfig })}
```

Comment:
We get an error when these components are left in on radial plots. I think it makes sense with the recharts API, because they list specific child components that can be used with each plot, and stuff like `PiePlot` does _not_ have x-axis/y-axis as valid children.